test/benchmarks/src/main/scala/scala/collection/immutable/VectorMapBenchmark.scala

```diff
@@ -0,0 +1,40 @@
+package scala.collection.immutable
+
+import org.openjdk.jmh.annotations._
+import org.openjdk.jmh.infra._
+import org.openjdk.jmh.runner.IterationType
+import benchmark._
+import java.util.concurrent.TimeUnit
+
+import scala.collection.mutable
+
+@BenchmarkMode(Array(Mode.AverageTime))
+@Fork(2)
+@Threads(1)
+@Warmup(iterations = 10)
+@Measurement(iterations = 10)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Benchmark)
+class VectorMapBenchmark {
+  @Param(Array("10", "100", "1000"))
+  var size: Int = _
+
+  var values: Vector[Any] = _
+
+  @Setup(Level.Trial) def initKeys(): Unit = {
+    values = (0 to size).map(i => (i % 10) match {
+      case 0 => i.toString
+      case 1 => i.toByte
```

Comment:
This seems excessive.  Object and non-object are different paths, and maybe constant-box vs. newly allocated box (byte vs double) is good to check.  Having to worry about so many seems excessive.